### PR TITLE
[PostgreSQL] Add new field id_columns in sync rules to generate id and adding support for views

### PR DESCRIPTION
**Pull request by parthpuri-elastic**
## Part of https://github.com/elastic/connectors-py/issues/1761

Added new field `id_columns` in advanced sync rules to ingest tables which do not have a primary key. So, now connector uses `id_columns` to generate unique ids. Also using `id_columns` users is able to fetch data of views.

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference
- [ ] if you added or changed Rich Configurable Fields for a Native Connector, you made a corresponding PR in [Kibana](https://github.com/elastic/kibana/blob/main/packages/kbn-search-connectors/types/native_connectors.ts)

## Release Note

Added a new optional field in PostgreSQL advanced sync rules `id_columns`. The user needs to pass the name of unique fields of the table in this field, and it will be used to generate a unique ID for documents. Users are able to index data of views by using advanced sync rules and `id_columns`.

### CODE REVIEW

**Comment by parthpuri-elastic** [comment]
buildkite test this

File: connectors/sources/postgresql.py [hunk 1]
```diff
@@ -83,6 +83,7 @@ class PostgreSQLAdvancedRulesValidator(AdvancedRulesValidator):
         "properties": {
             "tables": {"type": "array", "minItems": 1},
             "query": {"type": "string", "minLength": 1},
+            "id_columns": {"type": "array", "minItems": 0},
```
Comment (artem-shelkovnikov) [inline connectors/sources/postgresql.py:86]: Why `minItems: 0`? So it's not required, right?

**Review by artem-shelkovnikov — APPROVED** [review]

File: connectors/sources/postgresql.py (hunk 1 above)
Comment (parthpuri-elastic) [reply connectors/sources/postgresql.py]: No, we need to keep `minItems: 1` for `id_columns` as its optional field, but if the user added that, we need at least one column to generate IDs. I have updated the code. Thanks.

**Comment by parthpuri-elastic** [comment]
buildkite test this

**Review by artem-shelkovnikov — APPROVED** [review]

**Comment by praveen-elastic** [comment]
Buildkite test this

**Pull request merged**